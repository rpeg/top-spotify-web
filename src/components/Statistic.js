import React from 'react';

const Statistic = ({ title, children }) => (
  <div>
    {title && (
      <div className="stat-title" style={{ textAlign: 'start' }}>
        <p>{title}</p>
      </div>
    )}
    <div className="stat-content">{children}</div>
  </div>
);

export default Statistic;
