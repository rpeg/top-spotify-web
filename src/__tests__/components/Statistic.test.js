import React from 'react';
import { shallow } from 'enzyme';

import Statistic from '../../components/Statistic';

describe('Statistic', () => {
  it('should not log errors in console', () => {
    const wrapper = shallow(<Statistic />);
    const spy = jest.spyOn(global.console, 'error');
    expect(wrapper).not.toBeNull();
    expect(spy).not.toHaveBeenCalled();
  });

  it('renders correctly without title', () => {
    const wrapper = shallow(<Statistic />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.stat-title').exists()).toBe(false);
  });

  it('renders correctly with title', () => {
    const wrapper = shallow(<Statistic title="abc" />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.stat-title').exists()).toBe(true);
  });
});
