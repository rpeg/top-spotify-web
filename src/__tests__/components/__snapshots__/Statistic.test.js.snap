// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statistic renders correctly with title 1`] = `
<div>
  <div
    className="stat-title"
    style={
      Object {
        "textAlign": "start",
      }
    }
  >
    <p>
      abc
    </p>
  </div>
  <div
    className="stat-content"
  >
    <Component />
  </div>
</div>
`;

exports[`Statistic renders correctly without title 1`] = `
<div>
  <div
    className="stat-content"
  >
    <Component />
  </div>
</div>
`;
